go/oasis-node/cmd/debug/beacon/beacon.go

```diff
@@ -0,0 +1,97 @@
+// Package beacon implements the beacon introspection debug sub-commands.
+package beacon
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+	"google.golang.org/grpc"
+
+	beacon "github.com/oasisprotocol/oasis-core/go/beacon/api"
+	"github.com/oasisprotocol/oasis-core/go/common/logging"
+	consensus "github.com/oasisprotocol/oasis-core/go/consensus/api"
+	cmdCommon "github.com/oasisprotocol/oasis-core/go/oasis-node/cmd/common"
+	cmdGrpc "github.com/oasisprotocol/oasis-core/go/oasis-node/cmd/common/grpc"
+)
+
+var (
+	beaconCmd = &cobra.Command{
+		Use:   "beacon",
+		Short: "debug the random beacon",
+	}
+
+	beaconStatusCmd = &cobra.Command{
+		Use:   "status",
+		Short: "query beacon status",
+		Run:   doBeaconStatus,
+	}
+
+	logger = logging.GetLogger("cmd/debug/beacon")
+)
+
+func doConnect(cmd *cobra.Command) (*grpc.ClientConn, beacon.Backend) {
+	if err := cmdCommon.Init(); err != nil {
+		cmdCommon.EarlyLogAndExit(err)
+	}
+
+	conn, err := cmdGrpc.NewClient(cmd)
+	if err != nil {
+		logger.Error("failed to establish connection with node",
+			"err", err,
+		)
+		os.Exit(1)
+	}
+
+	client := beacon.NewBeaconClient(conn)
+
+	return conn, client
+}
+
+func doBeaconStatus(cmd *cobra.Command, args []string) {
+	conn, client := doConnect(cmd)
+	defer conn.Close()
+
+	logger.Info("querying latest beacon")
+
+	b, err := client.GetBeacon(context.Background(), consensus.HeightLatest)
+	if err != nil {
+		logger.Error("failed to query beacon",
+			"err", err,
+		)
+		os.Exit(1)
+	}
+
+	fmt.Printf("beacon: %x", b)
```

Comment:
I guess it would be nicer if this was part of the JSON output? You could define a structure that includes the current beacon and the PVSS state and dump that instead.

Reply:
The command is a debug one, if people are interested in programmatically interacting with the beacon they should use gRPC, not fork/exec.

Reply:
I changed the output, against my better judgement.